client/src/app/shared/shared-moderation/user-moderation-dropdown.component.ts

```diff
@@ -271,20 +273,20 @@ export class UserModerationDropdownComponent implements OnInit, OnChanges {
     return account && this.authService.getUser().account.id === account.id
   }
 
-  private buildActions () {
-    this.userActions = []
+  private async buildActions () {
+    const userActions: DropdownAction<{ user: User, account: AccountMutedStatus }>[][] = []
 
     if (this.prependActions && this.prependActions.length !== 0) {
-      this.userActions = [
-        this.prependActions
-      ]
+      userActions.push(this.prependActions)
     }
 
     const myAccountModerationActions = this.buildMyAccountModerationActions()
     const instanceModerationActions = this.buildInstanceModerationActions()
 
-    if (myAccountModerationActions.length !== 0) this.userActions.push(myAccountModerationActions)
-    if (instanceModerationActions.length !== 0) this.userActions.push(instanceModerationActions)
+    if (myAccountModerationActions.length !== 0) userActions.push(myAccountModerationActions)
+    if (instanceModerationActions.length !== 0) userActions.push(instanceModerationActions)
+
+    this.userActions = await this.hooks.wrapObject(userActions, 'admin-users', 'filter:admin-user-moderation.actions.create.result')
```

Comment:
The user moderation dropdown can be called in many pages. So I think you can create a `moderation` scope and use `filter:user-moderation.actions.create.result` as hook name